src/lib.rs

```diff
@@ -467,10 +509,88 @@ where
     self.r_U_secondary = r_U_secondary;
     self.r_W_secondary = r_W_secondary;
 
+    self.ri_primary = r_next_primary;
+    self.ri_secondary = r_next_secondary;
+
     Ok(())
   }
 
-  /// Verify the correctness of the `RecursiveSNARK`
+  /// add a randomizing fold to a `RecursiveSNARK`
+  /// if you plan on using a randomized CompressedSNARK, you don't need to call this
+  pub fn randomizing_fold(
```

Comment:
Does this need to be public?

Reply:
I think probably not. There's no use case where a user might want _just_ the randomized (uncompressed) IVC, right?

Reply:
There may be some use-case in the future where one generates a randomized IVC proof on the client-side and sends it to a server for compression. Perhaps, for this PR, we can ignore this use-case to keep the main interface exposed to the outside simple and small.